File: app/src/main/java/com/perflyst/twire/misc/RoundImageAnimation.java
package com.perflyst.twire.misc;

import android.graphics.Bitmap;
import android.view.animation.Animation;
import android.view.animation.Transformation;
import android.widget.ImageView;

import com.perflyst.twire.service.Service;

/**
 * Created by Sebastian Rask on 24-06-2016.
 */
public class RoundImageAnimation extends Animation {
    private final int fromRounded, toRounded;
    private final ImageView view;
    private final Bitmap imageBitmap;

    public RoundImageAnimation(int fromRounded, int toRounded, ImageView view, Bitmap imageBitmap) {
        this.fromRounded = fromRounded;
        this.toRounded = toRounded;
        this.view = view;
        this.imageBitmap = imageBitmap;
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        int rounded = (int) (fromRounded + (toRounded - fromRounded) * interpolatedTime);
        view.setImageBitmap(Service.getRoundedCornerBitmap(imageBitmap, rounded));
        view.requestLayout();
    }

    @Override
    public void initialize(int width, int height, int parentWidth, int parentHeight) {
        super.initialize(width, height, parentWidth, parentHeight);
    }

    @Override
    public boolean willChangeBounds() {
        return false;
    }
}

File: app/src/main/java/com/perflyst/twire/misc/RoundImageAnimation.kt
package com.perflyst.twire.misc

import android.graphics.Bitmap
import android.view.animation.Animation
import android.view.animation.Transformation
import android.widget.ImageView
import androidx.core.graphics.drawable.RoundedBitmapDrawable
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory

/**
 * Created by Sebastian Rask on 24-06-2016.
 */
class RoundImageAnimation(
    private val fromRounded: Int,
    private val toRounded: Int,
    private val view: ImageView,
    imageBitmap: Bitmap?,
) : Animation() {
    var drawable: RoundedBitmapDrawable =
        RoundedBitmapDrawableFactory.create(view.resources, imageBitmap)

    init {
        view.setImageDrawable(drawable)
    }

    override fun applyTransformation(interpolatedTime: Float, t: Transformation?) {
        val rounded = (fromRounded + (toRounded - fromRounded) * interpolatedTime)
        drawable.setCornerRadius(rounded)
        view.requestLayout()
    }

    override fun willChangeBounds(): Boolean {
        return false
    }
}
